IPython/terminal/pt_inputhooks/qt.py

```diff
@@ -1,10 +1,15 @@
 import sys
 from IPython.external.qt_for_kernel import QtCore, QtGui
 
+# If we create a QApplication, keep a reference to it so that it doesn't get
+# garbage collected.
+_appref = None
+
 def inputhook(context):
+    global _appref
     app = QtCore.QCoreApplication.instance()
     if not app:
-        return
```

Comment:
I think we broke the use case in 5.0, and I'm fine restoring the behavior in next minor/patch version {especially since LTS]
I don't think in the long run we should create the App if it does not exists. 

So I would reinstate it to not break user-code/habits. But warn about it, and remove it in the Py3 - Only branch.